### DISCUSSION TITLE
Getting DB connection error while server is running continuously

### DISCUSSION THREAD

**Discussion by ramkishore-yerubandi** [vendurehq/vendure#1652] [Q&A]
rcore-dev] [2022-07-06 10:53:46] error 7/6/22, 10:53 AM - [Vendure Server] remaining connection slots are reserved for non-replication superuser connections
[rcore-dev] [2022-07-06 10:53:46] error: remaining connection slots are reserved for non-replication superuser connections
[rcore-dev] [2022-07-06 10:53:46]     at Parser.parseErrorMessage (/workspace/node_modules/pg-protocol/dist/parser.js:287:98)
[rcore-dev] [2022-07-06 10:53:46]     at Parser.handlePacket (/workspace/node_modules/pg-protocol/dist/parser.js:126:29)
[rcore-dev] [2022-07-06 10:53:46]     at Parser.parse (/workspace/node_modules/pg-protocol/dist/parser.js:39:38)
[rcore-dev] [2022-07-06 10:53:46]     at TLSSocket.<anonymous> (/workspace/node_modules/pg-protocol/dist/index.js:11:42)
[rcore-dev] [2022-07-06 10:53:46]     at TLSSocket.emit (node:events:527:28)
[rcore-dev] [2022-07-06 10:53:46]     at TLSSocket.emit (node:domain:475:12)
[rcore-dev] [2022-07-06 10:53:46]     at addChunk (node:internal/streams/readable:315:12)
[rcore-dev] [2022-07-06 10:53:46]     at readableAddChunk (node:internal/streams/readable:289:9)
[rcore-dev] [2022-07-06 10:53:46]     at TLSSocket.Readable.push (node:internal/streams/readable:228:10)
[rcore-dev] [2022-07-06 10:53:46]     at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23) 
[

**Comment by michaelbromley** [comment]
Hi, what version of Vendure core does this occur on?

**Reply by ramkishore-yerubandi** [reply]
1.6.2

**Reply by michaelbromley** [reply]
ok. Can you try upgrading to 1.6.3 and see if it persists? We made some fixes to the way DB connections are handled internally.

**Reply by ramkishore-yerubandi** [reply]
Ok

**Reply by ramkishore-yerubandi** [reply]
Got this error again

[ecom] [2022-07-14 07:52:13] error 7/14/22, 7:52 AM - [Vendure Server] remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13] error: remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13]     at Parser.parseErrorMessage (/workspace/node_modules/pg-protocol/dist/parser.js:287:98)
[ecom] [2022-07-14 07:52:13]     at Parser.handlePacket (/workspace/node_modules/pg-protocol/dist/parser.js:126:29)
[ecom] [2022-07-14 07:52:13]     at Parser.parse (/workspace/node_modules/pg-protocol/dist/parser.js:39:38)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.<anonymous> (/workspace/node_modules/pg-protocol/dist/index.js:11:42)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:events:527:28)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:domain:475:12)
[ecom] [2022-07-14 07:52:13]     at addChunk (node:internal/streams/readable:315:12)
[ecom] [2022-07-14 07:52:13]     at readableAddChunk (node:internal/streams/readable:289:9)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.Readable.push (node:internal/streams/readable:228:10)
[ecom] [2022-07-14 07:52:13]     at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23) 
[ecom] [2022-07-14 07:52:13] error 7/14/22, 7:52 AM - [Vendure Server] remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13] error: remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13]     at Parser.parseErrorMessage (/workspace/node_modules/pg-protocol/dist/parser.js:287:98)
[ecom] [2022-07-14 07:52:13]     at Parser.handlePacket (/workspace/node_modules/pg-protocol/dist/parser.js:126:29)
[ecom] [2022-07-14 07:52:13]     at Parser.parse (/workspace/node_modules/pg-protocol/dist/parser.js:39:38)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.<anonymous> (/workspace/node_modules/pg-protocol/dist/index.js:11:42)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:events:527:28)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:domain:475:12)
[ecom] [2022-07-14 07:52:13]     at addChunk (node:internal/streams/readable:315:12)
[ecom] [2022-07-14 07:52:13]     at readableAddChunk (node:internal/streams/readable:289:9)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.Readable.push (node:internal/streams/readable:228:10)
[ecom] [2022-07-14 07:52:13]     at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23) 
[ecom] [2022-07-14 07:52:13] error 7/14/22, 7:52 AM - [Vendure Server] remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13] error: remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13]     at Parser.parseErrorMessage (/workspace/node_modules/pg-protocol/dist/parser.js:287:98)
[ecom] [2022-07-14 07:52:13]     at Parser.handlePacket (/workspace/node_modules/pg-protocol/dist/parser.js:126:29)
[ecom] [2022-07-14 07:52:13]     at Parser.parse (/workspace/node_modules/pg-protocol/dist/parser.js:39:38)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.<anonymous> (/workspace/node_modules/pg-protocol/dist/index.js:11:42)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:events:527:28)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:domain:475:12)
[ecom] [2022-07-14 07:52:13]     at addChunk (node:internal/streams/readable:315:12)
[ecom] [2022-07-14 07:52:13]     at readableAddChunk (node:internal/streams/readable:289:9)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.Readable.push (node:internal/streams/readable:228:10)
[ecom] [2022-07-14 07:52:13]     at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23) 
[ecom] [2022-07-14 07:52:13] error 7/14/22, 7:52 AM - [Vendure Server] remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13] error: remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13]     at Parser.parseErrorMessage (/workspace/node_modules/pg-protocol/dist/parser.js:287:98)
[ecom] [2022-07-14 07:52:13]     at Parser.handlePacket (/workspace/node_modules/pg-protocol/dist/parser.js:126:29)
[ecom] [2022-07-14 07:52:13]     at Parser.parse (/workspace/node_modules/pg-protocol/dist/parser.js:39:38)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.<anonymous> (/workspace/node_modules/pg-protocol/dist/index.js:11:42)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:events:527:28)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:domain:475:12)
[ecom] [2022-07-14 07:52:13]     at addChunk (node:internal/streams/readable:315:12)
[ecom] [2022-07-14 07:52:13]     at readableAddChunk (node:internal/streams/readable:289:9)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.Readable.push (node:internal/streams/readable:228:10)
[ecom] [2022-07-14 07:52:13]     at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23) 
[ecom] [2022-07-14 07:52:13] error 7/14/22, 7:52 AM - [Vendure Server] remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13] error: remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13]     at Parser.parseErrorMessage (/workspace/node_modules/pg-protocol/dist/parser.js:287:98)
[ecom] [2022-07-14 07:52:13]     at Parser.handlePacket (/workspace/node_modules/pg-protocol/dist/parser.js:126:29)
[ecom] [2022-07-14 07:52:13]     at Parser.parse (/workspace/node_modules/pg-protocol/dist/parser.js:39:38)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.<anonymous> (/workspace/node_modules/pg-protocol/dist/index.js:11:42)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:events:527:28)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:domain:475:12)
[ecom] [2022-07-14 07:52:13]     at addChunk (node:internal/streams/readable:315:12)
[ecom] [2022-07-14 07:52:13]     at readableAddChunk (node:internal/streams/readable:289:9)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.Readable.push (node:internal/streams/readable:228:10)
[ecom] [2022-07-14 07:52:13]     at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23) 
[ecom] [2022-07-14 07:52:13] error 7/14/22, 7:52 AM - [Vendure Server] remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13] error: remaining connection slots are reserved for non-replication superuser connections
[ecom] [2022-07-14 07:52:13]     at Parser.parseErrorMessage (/workspace/node_modules/pg-protocol/dist/parser.js:287:98)
[ecom] [2022-07-14 07:52:13]     at Parser.handlePacket (/workspace/node_modules/pg-protocol/dist/parser.js:126:29)
[ecom] [2022-07-14 07:52:13]     at Parser.parse (/workspace/node_modules/pg-protocol/dist/parser.js:39:38)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.<anonymous> (/workspace/node_modules/pg-protocol/dist/index.js:11:42)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:events:527:28)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.emit (node:domain:475:12)
[ecom] [2022-07-14 07:52:13]     at addChunk (node:internal/streams/readable:315:12)
[ecom] [2022-07-14 07:52:13]     at readableAddChunk (node:internal/streams/readable:289:9)
[ecom] [2022-07-14 07:52:13]     at TLSSocket.Readable.push (node:internal/streams/readable:228:10)
[ecom] [2022-07-14 07:52:13]     at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23)

**Reply by michaelbromley** [reply]
Can you give some context about the environment? What host, what database & version etc?

**Reply by ramkishore-yerubandi** [reply]
Hosted app in DigitalOcean App Platform and Postgres DB in DigitalOcean itself with Basic / 1 vCPU / 1 GB RAM / 10 GB SSD  / Connection limit: 22

**Comment by dylviz** [comment]
I know this is old but I also ran into this issue recently. The problem was fixed once I switched to using a DB connection pool rather than directly connecting. Hope this helps someone in the future.
